data/state/accountsDB.go

```diff
@@ -20,6 +20,9 @@ type AccountsDB struct {
 	marshalizer    marshal.Marshalizer
 	accountFactory AccountFactory
 
+	codeForEviction map[string]struct{}
```

Comment:
codeForEviction cannot be used with current state of smart contract process, as delete function, delete smart contract is not called at all.

Reply:
If a code is updated and the old version is not used anymore, it needs to be evicted.